services/repository/src/main/java/org/sagebionetworks/repo/web/service/SearchServiceImpl.java

```diff
@@ -156,13 +158,17 @@ public String createQueryString(UserInfo userInfo, SearchQuery searchQuery)
 	 * @param searchQuery
 	 * @return
 	 */
-	public String filterSeachForAuthorization(UserInfo userInfo,String searchQuery) {
+	public static String filterSeachForAuthorization(UserInfo userInfo,String searchQuery) {
 		if(userInfo == null) throw new IllegalArgumentException("UserInfo cannot be null");
 		if (!userInfo.isAdmin()) {
-			StringBuilder builder = new StringBuilder(searchQuery);
-			builder.append("&");
-			builder.append(SearchHelper.formulateAuthorizationFilter(userInfo));
-			searchQuery = builder.toString();
+			String[] splitQuery = searchQuery.split("q=");
```

Comment:
modify the object model rather than generate search followed by parse or search.